config/v1/types_infrastructure.go

```diff
@@ -354,6 +354,40 @@ type AWSPlatformSpec struct {
 	// There must be only one ServiceEndpoint for a service.
 	// +optional
 	ServiceEndpoints []AWSServiceEndpoint `json:"serviceEndpoints,omitempty"`
+
+	// ResourceTags is a list of additional tags to apply to AWS resources created for the cluster.
+	// See https://docs.aws.amazon.com/general/latest/gr/aws_tagging.html for information on tagging AWS resources.
+	// AWS supports a minimum of 10 tags per resource. OpenShift reserves 5 tags for its use, leaving 5 tags
+	// available for the user.
+	// ResourceTags field is mutable and items can be removed.
+	// When a tag is removed from this list, management of the tag is stopped.
+	// The tag will remain in an unmanaged state on any existing resource.
+	// +kubebuilder:validation:MaxItems=5
+	// +optional
+	ResourceTags []AWSResourceTagSpec `json:"resourceTags,omitempty"`
+}
+
+// AWSResourceTagSpec is a tag to apply to AWS resources created for the cluster.
+type AWSResourceTagSpec struct {
+	// key is the key of the AWS tag. The value must consist only of alphanumeric and certain special characters(_.:/=+-@\\).
+	// key should not start with "openshift.io" or "kubernetes.io". Both of these prefixes are reserved for use by the platform.
+	// key must be no more than 128 characters in length.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=128
+	// +kubebuilder:validation:Pattern=`^[0-9A-Za-z_.:/=+-@]+$`
```

Comment:
A note for other reviewers:

This pattern is the best we can do for now. We cannot exclude strings using openapi regexes so must do so via an admission plugin, this is being implemented in https://github.com/openshift/kubernetes/pull/1232

Reply:
adding an admission plugin directly to openshift/kubernetes for infrastructure is allowed because it has to be present before we have the ability to have an admission webhook